src/Framework/ContaoFramework.php

```diff
@@ -472,4 +480,30 @@ private function canSkipTokenCheck(): bool
             || false === $this->request->attributes->get('_token_check')
         ;
     }
+
+    /**
+     * Build hooks globals.
+     */
+    private function buildHookGlobals(): void
+    {
+        foreach ($this->hookListeners as $hookName => $priorities) {
```

Comment:
I thought I already mentioned that somewhere, but can't find it. These lines can be much simplified like this:

```php
foreach ($this->hookListeners as $hookName => $priorities) {
    if (isset($GLOBALS['TL_HOOKS'][$hookName]) && is_array($GLOBALS['TL_HOOKS'][$hookName])) {
        if (isset($priorities[0]) {
            $priorities[0] = array_merge($GLOBALS['TL_HOOKS'][$hookName], $priorities[0]);
        } else {
            $priorities[0] = $GLOBALS['TL_HOOKS'][$hookName];
        }
    }

    $GLOBALS['TL_HOOKS'][$hookName] = array_merge_recursive($priorities);
}
```

Reply:
You did, but it's still not true. `array_merge_recursive` with one argument will do nothing. Or do you mean `array_merge_recursive(...$priorities)`?

Reply:
Sorry, the correct call would be `call_user_func_array('array_merge', $priorities)`

Reply:
Okay I added your suggest change in https://github.com/contao/core-bundle/pull/1094/commits/863f8137daf27824b665faa29a0ed1840b9160d9